core/src/data/dataSource.cpp

```diff
@@ -38,19 +43,7 @@ void DataSource::setTileData(const TileID& _tileID, const std::shared_ptr<TileDa
         m_tileStore[_tileID] = _tileData;
 }
 
-void DataSource::setUrlTemplate(const std::string& _urlTemplate){
```

Comment:
I couldn't think of an instance in which we would be modifying a data source that way, especially since they'll be defined statically in the style sheet. And as you say, it's not costly to create a new instance should we need it.